gossip/src/cluster_info.rs

```diff
@@ -929,13 +929,21 @@ impl ClusterInfo {
         &self,
         transaction: &Transaction,
         tpu: Option<SocketAddr>,
+        connection_cache: &Arc<ConnectionCache>,
     ) -> Result<(), GossipError> {
         let tpu = tpu
             .map(Ok)
-            .unwrap_or_else(|| self.my_contact_info().tpu(contact_info::Protocol::UDP))?;
+            .unwrap_or_else(|| self.my_contact_info().tpu(connection_cache.protocol()))?;
         let buf = serialize(transaction)?;
-        self.socket.send_to(&buf, tpu)?;
-        Ok(())
```

Comment:
How does UDP connection cache compare with this code in terms of performance?
We are not going to make UDP path slower by switching to UDP connection cache, right?

Reply:
The UDP connection cache/ udp client is thin wrapper over socket as udp really is connectionless. We just return the dummy from the cache and use it to send. There will be some extra instructions to get it from the cache -- that is negligible in the context of cost of sending the packets.